components/select/src/single-select-a11y/single-select-a11y.js

```diff
@@ -0,0 +1,392 @@
+                onClose={closeMenu}
+                onEndReached={onEndReached}
+                onFilterChange={onFilterChange}
+            />
+        </div>
+    )
+}
+
+SingleSelectA11y.propTypes = {
+    /** necessary for IDs that are required for accessibility **/
+    idPrefix: PropTypes.string.isRequired,
+
+    /** An array of options **/
+    options: PropTypes.arrayOf(optionProp).isRequired,
+
+    /** A callback that will be called with the new value or an empty string **/
+    onChange: PropTypes.func.isRequired,
+
+    /** Will focus the select initially **/
+    autoFocus: PropTypes.bool,
+
+    /** Additional class names that will be applied to the root element **/
+    className: PropTypes.string,
+
+    /** This will allow us to put an aria-label on the clear button **/
+    clearText: requiredIf((props) => props.clearable, PropTypes.string),
+
+    /** Whether a clear button should be displayed or not **/
+    clearable: PropTypes.bool,
+
+    /** Allows to override what's rendered inside the `button[role="option"]`.
+     * Can be overriden on an individual option basis **/
+    customOption: PropTypes.elementType,
+
+    /** A value for a `data-test` attribute on the root element **/
+    dataTest: PropTypes.string,
+
+    /** Renders a select with lower height **/
+    dense: PropTypes.bool,
+
+    /** Disables all interactions with the select (except focussing) **/
+    disabled: PropTypes.bool,
+
+    /** Text or component to display when there are no options **/
+    empty: PropTypes.node,
+
+    /** Applies 'error' appearance for validation feedback. Mutually exclusive with `warning` and `valid` props **/
+    error: sharedPropTypes.statusPropType,
+
+    /** Help text that will be displayed below the input **/
+    filterHelpText: PropTypes.string,
+
+    /** Value will be used as aria-label attribute on the filter input **/
+    filterLabel: PropTypes.string,
+
+    /** Placeholder for the filter input **/
+    filterPlaceholder: PropTypes.string,
+
+    /** Value of the filter input **/
+    filterValue: PropTypes.string,
+
+    /** Whether the select should display a filter input **/
+    filterable: PropTypes.bool,
+
+    /** Should contain the id of the element that labels the select, if applicable **/
+    labelledBy: PropTypes.string,
+
+    /** Will show a loading indicator at the end of the options-list **/
+    loading: PropTypes.bool,
+
+    /** Text that will be displayed next to the loading indicator **/
+    menuLoadingText: PropTypes.string,
+
+    /** Allows to modify the max height of the menu **/
+    menuMaxHeight: PropTypes.string,
+
+    /** String that will be displayed when the select is being filtered but the options array is empty **/
+    noMatchText: requiredIf((props) => props.filterable, PropTypes.string),
+
+    /** For a11y: How aggressively the user should be updated about changes in options **/
+    optionUpdateStrategy: PropTypes.oneOf(['off', 'polite', 'assertive']),
```

Comment:
Is this only used for `aria-live`? If so could we at least mention that's what it controls? Or even a reference to https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-live

Reply:
Fair point! I'm having difficulties with finding a good name though.. Just `ariaLive` or `ariaLiveForOptions`? What do you think